pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kanctl
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+)
+
+const (
+	tlsSecretFlag                  = "tls"
+	repoServerUserAccessSecretFlag = "repoServerUserAccess"
+	repoAdminUserSecretFlag        = "repoAdminUser"
+	repoPasswordSecretFlag         = "repoPassword"
+	repoServerUserFlag             = "repoServerUser"
+	repoServerAdminUserFlag        = "repoServerAdminUser"
+	s3LocationCredsSecretFlag      = "s3LocationCreds"
+	s3LocationSecretFlag           = "s3Location"
+	defaultKanisterNamespace       = "kanister"
+	defaultRepositoryServerHost    = "localhost"
```

Comment:
for the flags/fields that we are expecting the values to be secret, should we change the name to include secret. For example, `repoPassword` -> `repoPasswordSecret`?

Reply:
Done !